modules/snmp-dest/snmpdest.c

```diff
@@ -0,0 +1,899 @@
+          self->session.contextEngineID == NULL)
+        {
+          self->session.contextEngineID =
+            snmpv3_generate_engineID(&self->session.contextEngineIDLen);
+        }
+
+      /*
+       * set boots and time, which will cause problems if this
+       * machine ever reboots and a remote trap receiver has cached our
+       * boots and time...  I'll cause a not-in-time-window report to
+       * be sent back to this machine.
+       */
+      if (self->session.engineBoots == 0)
+        self->session.engineBoots = 1;
+      if (self->session.engineTime == 0)    /* not really correct, */
+        self->session.engineTime = get_uptime();  /* but it'll work. Sort of. */
+    }
+
+  self->ss = snmp_add(&self->session,
+                      netsnmp_transport_open_client("snmptrap", self->session.peername),
+                      NULL, NULL);
+  if (self->ss == NULL)
+    {
+      for (i = 0; i < argc; ++i)
+        g_free(args[i]);
+      free_session(&self->session);
+      return FALSE;
+    }
+
+  self->session_initialized = TRUE;
+  return TRUE;
+}
+
+static gboolean
+snmpdest_dd_init(LogPipe *s)
+{
+  SNMPDestDriver *self = (SNMPDestDriver *)s;
+
+  if (!log_dest_driver_init_method(s))
+    return FALSE;
+
+  /* prepare the host:port format - the session.remote_port doesn't work */
+  self->host_port = g_string_sized_new(64);
+  if (self->transport)
+    g_string_append_printf(self->host_port, "%s:", self->transport);
+  g_string_append_printf(self->host_port, "%s:%d", self->host, self->port);
+
+  self->queue = log_dest_driver_acquire_queue(&self->super, snmpdest_dd_format_persist_name(self));
+
+  stats_lock();
+  stats_cluster_logpipe_key_set(&sc_key_queued, SCS_SNMP, snmpdest_dd_format_stats_instance(self), NULL );
+  stats_register_counter(1, &sc_key_queued, SC_TYPE_QUEUED, &self->queued_messages);
+
+  stats_cluster_logpipe_key_set(&sc_key_dropped, SCS_SNMP, snmpdest_dd_format_stats_instance(self), NULL );
+  stats_register_counter(1, &sc_key_dropped, SC_TYPE_DROPPED, &self->dropped_messages);
+
+  stats_cluster_logpipe_key_set(&sc_key_processed, SCS_SNMP, snmpdest_dd_format_stats_instance(self), NULL );
+  stats_register_counter(1, &sc_key_processed, SC_TYPE_PROCESSED, &self->processed_messages);
+  stats_unlock();
+
+  msg_verbose("Initializing SNMP destination",
+              evt_tag_str("host", self->host),
+              evt_tag_int("port", self->port),
+              NULL);
+
+  GlobalConfig *cfg = log_pipe_get_config(s);
+  gchar **tz = &self->template_options.time_zone[LTZ_LOCAL],
+          *time_zone = cfg->template_options.time_zone[LTZ_SEND];
+  if (!*tz && time_zone)
+    /* inherit the global set_time_zone if the local one hasn't been set */
+    *tz = g_strdup(time_zone);
+  log_template_options_init(&self->template_options, cfg);
+  /* set the cfg field of the log template items */
+  GList *t = self->snmp_templates;
+  while (t)
+    {
+      ((LogTemplate *)t->data)->cfg = cfg;
```

Comment:
Do we need to set this again? We used `template = log_template_new(cfg, NULL);` during register, that theoretically does this.

Reply:
you are right. I removed this code part.